### ISSUE TITLE
maxseqlength is not supported by uchime_denovo command

### ISSUE THREAD

**Issue by petaripenev** [torognes/vsearch#544]
My version is **v2.13.3_linux_x86_64**. Sorry if this has already been patched with a newer version.

Minimal code to reproduce error:

```
printf '>seq\n%*s' 11 | tr ' ' 'A' | vsearch --uchime_denovo - --uchimeout test.txt --maxseqlength 300000
```

Results in the following error:

```
Fatal error: Invalid options to command uchime_denovo
Invalid option(s): --maxseqlength
The valid options for the uchime_denovo command are: --abskew --alignwidth --borderline --chimeras --dn --fasta_score --fasta_width --gapext --gapopen --hardmask --log --match --mindiffs --mindiv --minh --mismatch --no_progress --nonchimeras --notrunclabels --qmask --quiet --relabel --relabel_keep --relabel_md5 --relabel_sha1 --sizein --sizeout --threads --uchimealns --uchimeout --uchimeout5 --xee --xn --xsize
```

I am not sure if there is a reason to limit uchime_denovo from being able to be ran on longer sequences, but chimeras could result in large contigs, so it makes sense to allow that.

**Comment by frederic-mahe** [comment]
I can confirm that  `--uchime_denovo` does not accept option `--maxseqlength` in vsearch's current version.

**Comment by torognes** [comment]
You're right. The `maxseqlength` option is missing for `uchime_denovo` and the other chimera detection commands.

I'll add it soon.

**Comment by torognes** [comment]
The current limit is 50 000 nucleotides.

I am not sure how well the `uchime_denovo` algorithm works on very long sequences. We are working on new algorithms for improved chimera detection in long (and accurate) sequences.

**Comment by torognes** [comment]
Fixed in commit 46c87a2.

The `maxseqlength` and the `minseqlength` options are now enabled for all chimera detection commands.

**Comment by frederic-mahe** [comment]
added regression tests to our test-suite (see https://github.com/frederic-mahe/vsearch-tests/commit/c86aedb6e1698c95d53fdb3031d25ec51228a5a6)
